File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.java
class Solution {
    public int subsetXORSum(int[] nums) {
       
        return func(nums,0,0);
    }
    int func(int arr[],int ind,int ans){
       
        if(ind==arr.length){
            return ans;
        }
        return func(arr,ind+1,ans^arr[ind])+func(arr,ind+1,ans);
    }
}

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.cs
public class Solution {
    public int SubsetXORSum(int[] nums) {
        
        return rec(nums,0,0);
    }
    
    public int rec(int[] nums, int ind , int ans)
    {
        if(ind == nums.Length)
        {
            return ans;
        }
        
        return rec(nums,ind+1,ans^nums[ind])+rec(nums,ind+1,ans);
    }
}